src/modules/menu/components/toolboxRight/Toggle3dButton.vue

```diff
@@ -1,28 +1,83 @@
 <template>
     <button
+        ref="toggle3dButton"
         class="toolbox-button"
         type="button"
-        :class="{ active: isActive }"
-        :title="$t(`tilt3d_${isActive ? 'active' : 'inactive'}`)"
+        :class="{ active: isActive, disabled: !webGlIsSupported }"
         data-cy="3d-button"
         @click="toggle3d"
     >
-        <img class="toolbox-button-icon" src="../../assets/button_3d.png" alt="3D" />
+        <FontAwesomeIcon :icon="['fas', 'cube']" flip="horizontal" />
     </button>
 </template>
 <script>
+import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
+import tippy from 'tippy.js'
 import { mapActions, mapState } from 'vuex'
 
+/**
+ * Returns true if WebGL is supported by the browser / hardware.
+ *
+ * Copied from https://developer.mozilla.org/en-US/docs/Web/API/WebGL_API/By_example/Detect_WebGL
+ *
+ * @returns {boolean}
+ */
+function checkWebGlSupport() {
+    // Create canvas element. The canvas is not added to the
+    // document itself, so it is never displayed in the
+    // browser window.
+    const canvas = document.createElement('canvas')
+
+    // Get WebGLRenderingContext from canvas element.
+    const gl = canvas.getContext('webgl') || canvas.getContext('experimental-webgl')
+
+    return gl instanceof WebGLRenderingContext
+}
+
 export default {
+    components: { FontAwesomeIcon },
+    data() {
+        return {
+            webGlIsSupported: false,
+        }
+    },
     computed: {
         ...mapState({
             isActive: (state) => state.ui.showIn3d,
         }),
+        tooltipContent() {
+            if (this.webGlIsSupported) {
+                return this.$t(`tilt3d_${this.isActive ? 'active' : 'inactive'}`)
+            }
+            return this.$t('3d_render_error')
+        },
+    },
+    watch: {
+        isActive() {
+            if (this.tooltip) {
+                this.tooltip.setContent(this.tooltipContent)
+            }
+        },
+    },
+    mounted() {
+        this.webGlIsSupported = checkWebGlSupport()
+        this.tooltip = tippy(this.$refs.toggle3dButton, {
+            theme: 'primary',
+            content: this.tooltipContent,
```

Comment:
Tooltip content will not be updated if the language changes.

Reply:
good catch, thanks